x/trigger/keeper/queue.go

```diff
@@ -0,0 +1,145 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/provenance-io/provenance/x/trigger/types"
+)
+
+// QueueTrigger Creates a QueuedTrigger and Enqueues it
+func (k Keeper) QueueTrigger(ctx sdk.Context, trigger types.Trigger) {
+	item := types.NewQueuedTrigger(trigger, ctx.BlockTime().UTC(), uint64(ctx.BlockHeight()))
+	k.Enqueue(ctx, item)
+}
+
+// QueuePeek Returns the next item to be dequeued.
+func (k Keeper) QueuePeek(ctx sdk.Context) (item types.QueuedTrigger) {
+	if k.QueueIsEmpty(ctx) {
+		panic("unable to peek empty queue")
+	}
+	index := k.getQueueStartIndex(ctx)
+	item = k.getQueueItem(ctx, index)
+	return
+}
```

Comment:
I feel like this should return a `*types.QueuedTrigger` instead and return `nil` if the queue is empty. Otherwise, in order to be safe, you would always have to call `k.QueueIsEmpty(ctx)` before calling `QueuePeek` (which then calls `k.QueueIsEmpty(ctx)` again).

Reply:
Sure, we can have it work like that.